main.go

```diff
@@ -274,6 +276,9 @@ func main() {
 
 	if grpcServerAddr != "" && grpcServerAddr != "0" {
 		server := internalgrpc.NewServer(grpcServerAddr)
+		// Register metrics service
+		metricService := metricservice.NewMetricsServiceServer(metricProvider)
+		api.RegisterProviderServiceServer(server.ServiceRegistrar(), metricService)
```

Comment:
suggest to wrap this to a method `RegisterTo` of `ProviderServer` and call `metricProviderServer.RegisterTo(server.ServiceRegistrar())` here